warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -84,12 +91,113 @@ public Entry<Key,Document> apply(@Nullable Entry<Key,Document> entry) {
         if (document.containsKey(PHRASE_INDEXES_ATTRIBUTE)) {
             Content content = (Content) document.get(PHRASE_INDEXES_ATTRIBUTE);
-            return PhraseIndexes.from(content.getContent());
-        } else {
+            PhraseIndexes phraseIndexes = PhraseIndexes.from(content.getContent());
+            phraseIndexMap = phraseIndexes.toMap();
+            allPhraseIndexes.addAll(phraseIndexes);
+        }
+        // now lets find all of the terms for excerpt fields and add them to the list
+        if (document.containsKey(JexlEvaluation.HIT_TERM_FIELD)) {
+            Attributes hitList = (Attributes) document.get(JexlEvaluation.HIT_TERM_FIELD);
+            // for each hit term
+            for (Attribute a : hitList.getAttributes()) {
+                ValueTuple hitTuple = attributeToHitTuple(a);
+                // if this is for a requested excerpt field
+                if (excerptFields.getFields().contains(hitTuple.getFieldName())) {
+                    // get the offset, preferring offsets that overlap with existing phrases for this field/eventId
+                    TermWeightPosition pos = getOffset(hitTuple, phraseIndexMap);
+                    if (pos != null) {
+                        // add the term as a phrase as defined in the term weight position. Note that this will collapse with any overlapping phrases already in
+                        // the list.
+                        allPhraseIndexes.addIndexTriplet(String.valueOf(hitTuple.getFieldName()), keyToEventId(a.getMetadata()), pos.getLowOffset(),
+                                        pos.getOffset());
+                    }
+                }
+            }
+        }
+        // return the file set of phrase indexes if any
+        return (allPhraseIndexes.isEmpty() ? null : allPhraseIndexes);
+    }
+    
+    /**
+     * Get the term weight position (offset) for the specified hit term. This will return an offset overlapping a phrase in the existing phrase index map first.
+     * Otherwise the first position will be returned.
+     * 
+     * @param hitTuple
+     *            The hit term tuple
+     * @param phraseIndexMap
+     *            The phrase indexes in map form to speed up searches.
+     * @return The TermWeightPosition for the given hit term
+     */
+    private TermWeightPosition getOffset(ValueTuple hitTuple, Map<String,Map<String,PhraseIndexes>> phraseIndexMap) {
+        Key docKey = hitTuple.getSource().getMetadata();
+        // if we do not know the source document key, then we cannot find the term offset
+        if (docKey == null) {
+            log.warn("Unable to find the source document for a hit term, skipping excerpt generation");
             return null;
         }
+        String fieldName = hitTuple.getFieldName();
+        String eventId = keyToEventId(docKey);
+        
+        // pull out the appropriate phrase indexes for this field/event id
+        PhraseIndexes phraseIndexes = null;
```

Comment:
This variable is not used anywhere outside of the try block, and doesn't need to be initialized here. You can perform a check for a pre-existing PhraseIndex where it's relevant at line 165, i.e.

```java
if(phraseIndexMap.containsKey(fieldName)) {
    PhraseIndexes phraseIndexes = phraseIndexMap.get(fieldName).get(eventId);
    for (int i = 0; i < twInfo.getTermOffsetCount(); i++) {
    ...
```

Reply:
not quite right because I still need to see of phraseIndexes is null as well, but since this is a hashmap lookup I can move it inside the loop.